ihmc-robotics-toolkit/src/test/java/us/ihmc/robotics/MatrixMissingToolsTest.java

```diff
@@ -313,4 +313,61 @@ public void testSetSelectedMatrixDiagonals()
          }
       }
    }
+
+   @Test
+   public void testPowerCalculation()
+   {
+      Random random = new Random(1738L);
+
+      int iters = 100;
+
+      for (int i = 0; i < iters; i++)
+      {
+         int matrixSize = random.nextInt(10);
+         DMatrixRMaj matrix = RandomMatrices_DDRM.rectangle(matrixSize, matrixSize, random);
+         DMatrixRMaj matrixPowerTest = new DMatrixRMaj(matrixSize, matrixSize);
+         int powerNumber = random.nextInt(15);
+
+         MatrixMissingTools.power(matrix, powerNumber, matrixPowerTest);
+
+         DMatrixRMaj matrixCompareTemporary = new DMatrixRMaj(matrixSize, matrixSize);
+         DMatrixRMaj matrixCompareResult = new DMatrixRMaj(matrix);
+
+         if (powerNumber > 1)
+         {
+            for (int j = 0; j < powerNumber - 1; j++)
+            {
+               CommonOps_DDRM.mult(matrix, matrixCompareResult, matrixCompareTemporary);
+               matrixCompareResult.set(matrixCompareTemporary);
+            }
```

Comment:
This is not really a test because you re-implemented the method here and compare it against the method, besides testing that the method runs properly it doesn't do much more.
To test I suggest the following:
- test simple examples
- test properties on the resulting matrix (try with identity, skew matrix, etc)
In general, testing requires some thinking about what the method should satisfy and the type of user errors that it should catch